cek/core/models.py

```diff
@@ -0,0 +1,346 @@
+        :ivar User user: Current user connected to the device. Can be different from context.user.
+    """
+
+    def __init__(self, session_dict):
+        self._session = session_dict
+        self._user = User(session_dict['user'])
+
+    @property
+    def id(self):
+        return self._session['sessionId']
+
+    @property
+    def is_new(self):
+        return self._session['new']
+
+    @property
+    def attributes(self):
+        return self._session.setdefault('sessionAttributes', {})
+
+    @property
+    def user(self):
+        return self._user
+
+
+class Device(object):
+    """Type which holds details about user's device.
+
+        :param dict device_dict: Dictionary represents the device field from the CEK request.
+
+        :ivar str id: ID of the device.
+    """
+
+    def __init__(self, device_dict):
+        self._device = device_dict
+
+    @property
+    def id(self):
+        return self._device['deviceId']
+
+
+class AudioPlayer(object):
+    """Type which holds details of the media content currently being played or played last.
+
+    :param dict audio_player_dict: Dictionary represents the AudioPlayer field from the CEK request.
+
+    :ivar num offset: is the most recent playback position of the recently played media in milliseconds.
+    :ivar num total: is the total duration of the recently played media in milliseconds.
+    :ivar str activity: is indicating the state of player. Can be "IDLE", "PLAYING", "PAUSED" or "STOPPED".
+    :ivar dict stream: contains details of the currently playing media. TODO: AudioStreamInfoObject specs are still WIP.
+    """
+
+    def __init__(self, audio_player_dict):
+        self._audio_player = audio_player_dict
+
+    @property
+    def offset(self):
+        return self._audio_player.get('offsetInMilliseconds', 0)
+
+    @property
+    def total(self):
+        return self._audio_player.get('totalInMilliseconds', 0)
+
+    @property
+    def activity(self):
+        return self._audio_player['playerActivity']
+
+    @property
+    def stream(self):
+        return self._audio_player.get('stream', None)
+
+
+class Context(object):
+    """Type which holds context information of the client.
+
+    :param dict context_dict: Dictionary represents the context from a CEK request.
+
+    :ivar AudioPlayer audio_player: holds details of media content currently being played or played last. Can be None if empty.
+    :ivar Device device: contains information of the client device.
+    :ivar User user: default User of the device.
+    """
+
+    def __init__(self, context_dict):
+        self._context = context_dict
+        self._user = User(context_dict['System']['user'])
+        self._device = Device(context_dict['device'])
+
+    @property
+    def audio_player(self):
+        if 'AudioPlayer' in self._context:
+            return AudioPlayer(self._context['AudioPlayer'])
```

Comment:
This will return a new instance every time the property is accessed. I think it's better to move this in the constructor, similar to user and device.

Reply:
Thanks, I missed this!